pkg/version/version.go

```diff
@@ -0,0 +1,105 @@
+package version
+
+import (
+	"context"
+	"errors"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v32/github"
+	"golang.org/x/oauth2"
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+	"gopkg.in/src-d/go-git.v4/storage/memory"
+)
+
+type Tag struct {
+	Name string    `json:"name"`
+	SHA  string    `json:"sha"`
+	Date time.Time `json:"date"`
+}
+
+// ListGitHubTags fetches git repo tags via GitHub API. It return tags in order.
+// Could support pagination. Return up to 30 by default
+func ListGitHubTags(owner, repo string) (*[]Tag, error) {
```

Comment:
Interesting that you return a pointer to a slice of Tags here. My first instinct would be to return `[]*Tag` so I'm really curious why you went with *[]Tag. 

Are you maybe ensuring that the underlying slice isn't duplicated? Here, you return a reference to the slice, rather than returning a new slice that points to the same underlying array right?